Ical.Net/DataTypes/Period.cs

```diff
@@ -138,34 +121,38 @@ public virtual IDateTime? EndTime
     /// A <see cref="Period"/> that has a date-only <see cref="StartTime"/>, no <see cref="EndTime"/>
     /// and no duration set, is considered to last for one day.
     /// </summary>
-    public virtual TimeSpan? Duration
-    {
-        get
-        {
-            if (_endTime == null && !StartTime.HasTime)
-            {
-                return TimeSpan.FromDays(1);
-            }
-            return _duration ?? _endTime?.Subtract(StartTime);
-        }
-        set
-        {
-            _duration = value;
-            if (_duration != null)
-            {
-                _endTime = null;
-            }
-        }
-    }
+    public virtual Duration? Duration { get; set; }
 
     /// <summary>
     /// Gets the original start time, end time, and duration as they were set,
     /// while <see cref="EndTime"/> or <see cref="Duration"/> properties may be calculated.
     /// </summary>
     /// <returns></returns>
-    public (IDateTime StartTime, IDateTime? EndTime, TimeSpan? Duration) GetOriginalValues()
+    public (IDateTime StartTime, IDateTime? EndTime, Duration? Duration) GetOriginalValues()
```

Comment:
Isn't this redundant now as these properties already represent the original values?

Reply:
yes, absolutely